spi/discovery/src/main/java/com/hazelcast/examples/spi/discovery/HostsDiscoveryStrategy.java

```diff
@@ -122,9 +125,6 @@
     }
 
     private boolean matchesDomain(String line) {
-        if (line.isEmpty()) {
```

Comment:
I think the left side here is the corrected one, otherwise the discovery failed under Ubuntu with an /etc/hosts with empty lines in it.